modules/lib/global.d.ts

```diff
@@ -1,210 +1,112 @@
-     * @param value Value to convert
+     * This globally available function will load a JavaScript file and return the exports as objects.
+     * The function implements parts of the `CommonJS Modules Specification`.
+     *
+     * @example
+     * // Require relative to this
+     * var other = require('./other.js');
+     *
+     * @example
+     * // Require absolute
+     * var other = require('/path/to/other.js');
+     *
+     * @example
+     * // Require without .js extension
+     * var other = require('./other');
+     *
+     * @param {string} path Path for javascript file (relative or absolute and .js ending is optional).
+     * @returns {object} Exports from loaded javascript.
+     * @global
      */
-    nullOrValue: <T = object>(value: T) => T | null | undefined;
+    declare const require: XpRequire;
 
     /**
-     * Doc registerMock.
+     * Resolves a path to another file. Can use relative or absolute path.
      *
-     * @param name Name of mock.
-     * @param value Value to register.
+     * @example
+     * // Resolve relative to this
+     * var path = resolve('./other.html');
+     *
+     * @example
+     * // Resolve absolute
+     * var path = resolve('/path/to/other.html');
+     *
+     * @param {string} path Path to resolve.
+     * @returns {*} Reference to an object.
+     * @global
      */
-    registerMock: (name: string, value: object) => void
-};
-
-declare type XpRequire = <Key extends keyof XpLibraries | string = string>(path: Key) =>
-    Key extends keyof XpLibraries ? XpLibraries[Key] : unknown;
-
-/**
- * This globally available function will load a JavaScript file and return the exports as objects.
- * The function implements parts of the `CommonJS Modules Specification`.
- *
- * @example
- * // Require relative to this
- * var other = require('./other.js');
- *
- * @example
- * // Require absolute
- * var other = require('/path/to/other.js');
- *
- * @example
- * // Require without .js extension
- * var other = require('./other');
- *
- * @param {string} path Path for javascript file (relative or absolute and .js ending is optional).
- * @returns {object} Exports from loaded javascript.
- * @global
- */
-declare const require: XpRequire;
+    declare const resolve: Resolve;
+}
 
-/**
- * Resolves a path to another file. Can use relative or absolute path.
- *
- * @example
- * // Resolve relative to this
- * var path = resolve('./other.html');
- *
- * @example
- * // Resolve absolute
- * var path = resolve('/path/to/other.html');
- *
- * @param {string} path Path to resolve.
- * @returns {*} Reference to an object.
- * @global
- */
-declare const resolve: (path: string) => {
-    getApplicationKey(): string;
-    getPath(): string;
-    getUri(): string;
-    isRoot(): boolean;
-    getName(): string;
-    getExtension(): string;
-};
+//──────────────────────────────────────────────────────────────────────────────
+// Making sure the file is a module
```

Comment:
Same goes for such comments. Keep it in the commit message. If it's super important to keep it here, one line of code would be enough.

Reply:
> Keep it in the commit message
Yes, I like keeping commit message, however commit messages can "disappear" since some people want to squash branches.

Reply:
I kept the comment, but removed the lines